### ISSUE TITLE
Various tests require stderr behavior CPython cannot and will not provide

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2405]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2405|
|Type|defect|
|Created|2007-01-16 16:50:36Z|

It is basically impossible for a CPython program to completely control its stderr output.  Tests which rely on this cannot be made to pass reliably.

Some tests which cannot deal with bytes on stderr:

twisted.test.test_unix.UnixSocketTestCase.testUncleanServerSocketLocking
twisted.test.test_unix.UnixSocketTestCase.testUncleanSocketLockingFromThePerspectiveOfAClientConnectingToTheDeadServerSocket
twisted.test.test_stdio.StandardInputOutputTestCase.testHostAndPeer
twisted.test.test_stdio.StandardInputOutputTestCase.testWriteSequence
twisted.test.test_stdio.StandardInputOutputTestCase.testWrite
twisted.test.test_stdio.StandardInputOutputTestCase.testLoseConnection
twisted.test.test_stdio.StandardInputOutputTestCase.testConsumer
twisted.test.test_stdio.StandardInputOutputTestCase.testProducer
twisted.test.test_util.GetPasswordTest.test_stdin

There may be others.

<details><summary>Searchable metadata</summary>

```
trac-id__2405 2405
type__defect defect
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1168966236000000 1168966236000000
changetime__1170272532000000 1170272532000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun removed owner|
|-|-|

Fixed in ignore-stderr-2405

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve set owner to @exarkun|
|-|-|

Mostly good. test_unix and test_util have been touched but did not get the usual cleanups.

Also, I was wondering how to prevent this from happening again ? Maybe add a note in coding standard ?

Anyway, when cleanups done, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [19537]) Merge ignore-stderr-2405

Author: exarkun
Reviewer: therve
Fixes [#2405](https://github.com/twisted/twisted/issues/2405)

Adjust tests which require Python child processes to be stderr-clean so that
they no longer require this.  Python makes it very difficult to guarantee no
random bytes will appear on stderr, so Twisted should neither depend on nor
promise this ability.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
